lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -112,23 +197,23 @@ def build_request_url(action, id = nil)
       end
 
       def success_from(response)
-        response == 'SUCCESS'
+        response.dig('operation', 'operationResult') == 'PENDING' || response.dig('operation', 'operationResult') == 'FAILED' || response.dig('operation', 'operationResult') == 'AUTHORIZED'
```

Comment:
❓ Question:

How does a 'FAILED' `operationResult` become a successful response ?

Reply:
The status of the gateway response for failed transactions is 200 so that why this were included this way

Reply:
🤔 mmm:

That answer re-inforces the issue, if the `operationResult` returns 'FAILED' the method is going to return true which is incorrect.

Reply:
I think @Heavyblade is correct here. The gateway returning `200` just tells us the request was successful, but the concern for `success_from` is to determine whether or not the transaction was successful.